lib/ansible/modules/network/nxos/nxos_facts.py

```diff
@@ -451,7 +451,12 @@ def populate(self):
 
     def populate_structured_interfaces(self, data):
         interfaces = dict()
-        for item in data['TABLE_interface']['ROW_interface']:
+        data = data['TABLE_interface']['ROW_interface']
+
+        if isinstance(data, dict):
+            data = [data]
```

Comment:
@r1k0 Can you import `to_list` and use that instead as it takes care of few other cases also?
```
from ansible.module_utils.network.common.utils import to_list

        data = data['TABLE_interface']['ROW_interface']

        if isinstance(data, dict):
            data = to_list(data)

        for item in data:
```